### PR TITLE
feat: Add base events aggregator

### PR DESCRIPTION
**Pull request by TheJarX**
TODO:

- [x] Use events API key (the one they send in the request body -> Now events are group by Api Key before they're sent
- [ ] Use an **actual** background job (?)
- [x] ~Discuss if we really need redis~
- [x] Add Docker file and deploy to staging
- [x] Add a README
- [ ] ~Ignore this project when publishing the library (?)~ shouldn't be an issue since we publish the output (`dist`)

### CODE REVIEW

**Comment by TheJarX** [comment]
This seem to be working. This version basically receives the events acting as proxy and stores them before sending them in digestible batches to the Mavis API. Pretty much the same as a CloudFlare worker. 

One thing to consider is that Mavis API won't reject the request if the body is missing something, it seems like the server just ignore them. So we should be careful with the requests sent to the Mavis API. With this aggregator we're checking that the request does have the `events` and `api_key` in the body.

File: mavis-aggregator/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    if (requestsInLastSecond >= MAX_REQUESTS_PER_SECOND) {
+        console.log('Rate limit reached, deferring flush');
+        setTimeout(flushEvents, FLUSH_INTERVAL);
+        return;
+    }
+
+    let eventsToSend: QueuedEvent[] = [];
+    let currentSize = 0;
+
+    while (true) {
```
Comment (julionav) [inline mavis-aggregator/src/index.ts]: I'm more or less sure the size limit is per request -- so this needs to happen AFTER grouping the events. 

Similar to the rate limit, it likely exist per API key

Comment (julionav) [reply mavis-aggregator/src/index.ts]: The fact that we are using TS doesn't mean we need to throw away what our lord .NET taught us. 

Keep this simple. Make a class for handling the event sending per api key

**Review by julionav — CHANGES_REQUESTED** [review]

**Comment by julionav** [comment]
We definetely don't need redis for this @TheJarX, so lets not add it. 
Few reasons why we don't:
- These are analytical events, so while they are important, it won't be the end of the world if some are lost for some reason

**Review by julionav — APPROVED** [review]
This is looking much better, nice job! Ship it

**Pull request merged**